src/csharp/Grpc.Core/CallCredentialsConfiguratorBase.cs

```diff
@@ -0,0 +1,38 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Collections.Generic;
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Base class for objects that can consume configuration from <c>CallCredentials</c> objects.
+    /// </summary>
+    public abstract class CallCredentialsConfiguratorBase
```

Comment:
Any reason this is an abstract class instead of an interface?

Reply:
Any reason to make it an interface instead of an abstract class? 😋 

You can't add new methods to an interface. Can we imagine a scenario where you would want to add new methods here?

Reply:
I suppose I'm too used to interfaces so the abstract base class seemed odd to me. But if serviceability is the reason, it seems fine.

Reply:
Yes, being able to add more CallCredential types is the reason for choosing abstract class.
Also more configuration args could be added for some of the credentials, in which case we would need to add a new method to the abstract class.